src/ORM/Query.php

```diff
@@ -710,6 +719,10 @@ public function __clone()
      */
     public function count()
     {
+        if (isset($this->_resultsCount)) {
+            return $this->_resultsCount;
+        }
```

Comment:
I think this needs to check the dirty state. For example:

```php
$query->where(['id >' => 2]);
echo $query->count();
$query->where(['id <' => 1]);
echo $query->count();
```

Would now be wrong. I think this needs to be `if (isset($this->_resultsCount) && !$this->_dirty) {`